src/Code.elm

```diff
@@ -0,0 +1,391 @@
+viewSyntaxSegments segments =
+    segments
+        |> NEL.map viewSegment
+        |> NEL.toList
+
+
+view : Code -> Html msg
+view code_ =
+    let
+        content =
+            case code_ of
+                Syntax segments ->
+                    viewSyntaxSegments segments
+
+                Builtin ->
+                    [ span [ class "builtin" ] [ text "Builtin" ] ]
+    in
+    pre [] [ code [] content ]
+
+
+
+-- JSON DECODE
+
+
+simpleSyntaxTypeFromString : String -> SyntaxType
+simpleSyntaxTypeFromString rawType =
+    case rawType of
+        "NumericLiteral" ->
+            NumericLiteral
+
+        "TextLiteral" ->
+            TextLiteral
+
+        "BytesLiteral" ->
+            BytesLiteral
+
+        "CharLiteral" ->
+            CharLiteral
+
+        "BooleanLiteral" ->
+            BooleanLiteral
+
+        "Blank" ->
+            Blank
+
+        "Var" ->
+            Var
+
+        "Constructor" ->
+            Constructor
+
+        "Request" ->
+            Request
+
+        "AbilityBraces" ->
+            AbilityBraces
+
+        "ControlKeyword" ->
+            ControlKeyword
+
+        "TypeOperator" ->
+            TypeOperator
+
+        "BindingEquals" ->
+            BindingEquals
+
+        "TypeAscriptionColon" ->
+            TypeAscriptionColon
+
+        "DataTypeKeyword" ->
+            DataTypeKeyword
+
+        "DataTypeParams" ->
+            DataTypeParams
+
+        "Unit" ->
+            Unit
+
+        "DataTypeModifier" ->
+            DataTypeModifier
+
+        "UseKeyword" ->
+            UseKeyword
+
+        "UsePrefix" ->
+            UsePrefix
+
+        "UseSuffix" ->
+            UseSuffix
+
+        "DelayForceChar" ->
+            DelayForceChar
+
+        "DelimiterChar" ->
+            DelimiterChar
+
+        "Parenthesis" ->
+            Parenthesis
+
+        "LinkKeyword" ->
+            LinkKeyword
+
+        "DocDelimiter" ->
+            DocDelimiter
+
+        "DocKeyword" ->
+            DocKeyword
+
+        _ ->
+            Blank
+
+
+decodeOp : Decode.Decoder SyntaxType
+decodeOp =
+    let
+        decodeOpTag =
+            at [ "annotation", "contents", "tag" ] Decode.string
+    in
+    Decode.map
+        Op
+        (Decode.oneOf
+            [ when decodeOpTag ((==) "Cons") (Decode.succeed Cons)
+            , when decodeOpTag ((==) "Snoc") (Decode.succeed Snoc)
+            , when decodeOpTag ((==) "Concat") (Decode.succeed Concat)
+            ]
+        )
+
+
+decodeHashQualifier : Decode.Decoder SyntaxType
+decodeHashQualifier =
+    let
+        decodeHashQualifierTag =
+            at [ "annotation", "contents", "tag" ] Decode.string
+
+        decodeNameOnly =
+            Decode.map NameOnly (at [ "annotation", "contents", "contents", "toText" ] Decode.string)
+
+        decodeHashOnly =
+            Decode.map HashOnly (at [ "annotation", "contents", "contents" ] Decode.string |> andThen Hash.decode)
```

Comment:
Technically this is a `ShortHash` object and not a string - I'm parsing it as a string, assuming we can make that backend change.